tests/Doctrine/Tests/OrmTestCase.php

```diff
@@ -168,4 +168,43 @@ protected function getSharedSecondLevelCacheDriverImpl(): Cache
 
         return $this->secondLevelCacheDriverImpl;
     }
+
+    /**
+     * @param array<mixed> $arguments
+     *
+     * @return mixed
+     */
+    public static function __callStatic(string $method, array $arguments)
+    {
+        if ($method === 'assertMatchesRegularExpression') {
+            self::assertRegExp(...$arguments);
```

Comment:
This one is already covered here: https://github.com/doctrine/orm/blob/e2e9f8fa97a2e8b6ad79f1b2f68c921262677cc1/tests/Doctrine/Tests/PHPUnit8PolyfilledTestCase.php#L19

That's super recent. Let's do similar things for `createStub`?

Reply:
why is this an extra test case? this would require to change all tests that need the polyfill, `OrmTestCase` is alrady the right abstraction that was added for this.

Reply:
The reasoning is in this comment: https://github.com/doctrine/orm/pull/8552#discussion_r607402830

TL;DR : having it in a separate class makes it easy to drop support for PHPUnit 8 once we can do so: we delete the class and remove the class alias.

In your case, since you are adding support for PHPUnit 7, I think you should actually introduce a `PHPUnit7PolyfilledTestCase` that extends `PHPUnit8PolyfilledTestCase`, put things that are missing on PHPUnit 7 in it (so `createStub`?), and add an if branch here: https://github.com/doctrine/orm/blob/e2e9f8fa97a2e8b6ad79f1b2f68c921262677cc1/tests/Doctrine/Tests/PolyfillableTestCase.php#L18-L20

Reply:
I do not think its a good argument, because you need to modify the base classes of many tests to change the polyfill. So its not really just deleting a file. Putting it into OrmTestCase isolates this code in one location and does not spread knowledge across the whole code base

Reply:
:thinking: the only test that changed its base class was `DoctrineTestCase`, see https://github.com/doctrine/orm/pull/8552/files

Reply:
Ah you are right, that makes sense i thought the hierachy was `PoliyfillTestCase` => `OrmTestCase`, but its `OrmTestCase` -> `DoctrineTestCase` -> `PolyfillTestCase`. That is better and i rework this.